copy/copy.go

```diff
@@ -1307,11 +1316,20 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		destStream = s
 		inputInfo.Digest = ""
 		inputInfo.Size = -1
+		uploadCompressorName = internalblobinfocache.Uncompressed
+		srcCompressorName = compressionFormat.Name()
 	} else {
 		// PreserveOriginal might also need to recompress the original blob if the desired compression format is different.
 		logrus.Debugf("Using original blob without modification")
 		compressionOperation = types.PreserveOriginal
 		inputInfo = srcInfo
+		if isCompressed {
+			uploadCompressorName = compressionFormat.Name()
+			srcCompressorName = compressionFormat.Name()
+		} else {
+			uploadCompressorName = internalblobinfocache.Uncompressed
+			srcCompressorName = internalblobinfocache.Uncompressed
+		}
```

Comment:
(It might be tiny bit simpler to set `srcCompressorName` based on `DetectCompression` universally, then this can just be `uploadCompressorName = srcCompressorName` and the three other cases above are also covered.

It would require setting the “unknown” values in the encrypted case separately, but being explicit about that could be a good thing.)

Reply:
Okay, changing it.